### ISSUE TITLE
Template No V-Show

### ISSUE THREAD

**Issue by martinmckenna** [vuejs/eslint-plugin-vue#1410]
**Please describe what the rule should do:**

Warn users that putting `v-show` directives on `<template />` tags does nothing.

Unlike `v-if`, `v-show` adds style attributes, which isn't valid for template tags.

**What category should the rule belong to?**
[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template v-show="someCondition" />
```
